### DISCUSSION TITLE
Feature Request - External Helm Deployments

### DISCUSSION THREAD

**Discussion by cberg-aot** [bcgov/quickstart-openshift#990] [Ideas]
See[ separating templating and deployment libraries](https://github.com/bcgov/nr-quickstart-typescript/discussions/989).  

Want: Ability to deploy external (artifacthub or private helm registry) charts as part of deploy backend.  

How: 
- Refactor action-deployer-openshift to accept helm generated templates?  
- Refactor repo to expect helm templates and not OC.yaml? (changes how templating is done, logic remains same)
- Other ideas? 

Route verification and other sanity tests are desired, but current logic is integrated with templating & OC expectations (ImageStreams & DeploymentConfigs not used in 3rd party charts, etc).

**Comment by DerekRoberts** [comment]
Helm support is on the roadmap!  I'm not familiar with Artifacthub.  Is it worthless, self-hosted garbage, like Artifactory?

**Comment by cberg-aot** [comment]
LOL 😂

It's essentially dockerhub for Helm charts.  Most Official Vendor charts
are located there.

This is all stemming from the Q&A question I outlined.  Contemplating
coupling the service deploy to the application pipeline since its a
dependant service.....

But this just SCREAMS separate project/deploy entirely to me as well

**Reply by DerekRoberts** [reply]
Okay, now I'm getting kinda psyched for the idea!  Helm is just sooooo much easier than K8s/oc templating.  Issue created.

https://github.com/bcgov/nr-quickstart-typescript/issues/993
